app/services/customers/update_service.rb

```diff
@@ -70,12 +72,26 @@ def update_currency(customer:, currency:, customer_update: false)
 
     private
 
-    def create_or_update_provider_customer(customer, billing_configuration = {})
-      handle_stripe_customer = customer.payment_provider.present?
-      handle_stripe_customer ||= (billing_configuration || {})[:provider_customer_id].present?
-      handle_stripe_customer ||= customer.stripe_customer&.provider_customer_id.present?
-      return unless handle_stripe_customer
+    def create_or_update_provider_customer(customer, payment_provider, billing_configuration = {})
+      handle_provider_customer = customer.payment_provider.present?
+      handle_provider_customer ||= (billing_configuration || {})[:provider_customer_id].present?
+
+      if payment_provider == 'stripe'
```

Comment:
Style/CaseLikeIf: Convert `if-elsif` to `case-when`. (https://rubystyle.guide#case-vs-if-else)